core/src/test/scala/org/apache/spark/util/AkkaUtilsSuite.scala

```diff
@@ -370,9 +371,13 @@ class AkkaUtilsSuite extends FunSuite with LocalSparkContext with ResetSystemPro
     val selection = slaveSystem.actorSelection(
       AkkaUtils.address(AkkaUtils.protocol(slaveSystem), "spark", "localhost", boundPort, "MapOutputTracker"))
     val timeout = AkkaUtils.lookupTimeout(conf)
-    intercept[TimeoutException] {
-      slaveTracker.trackerActor = Await.result(selection.resolveOne(timeout * 2), timeout)
+    val result = Try(Await.result(selection.resolveOne(timeout * 2), timeout))
+
+    assert(result.isFailure === true)
+    val exception = result match {
+      case Failure(ex) => ex
     }
```

Comment:
this will create a lot of warnings complaining that the match is not exhaustive. I thin you'll need to add a `case _ => fail(...)` to fix this